packages/lodestar/src/db/api/beacon/repositories/badBlock.ts

```diff
@@ -0,0 +1,30 @@
+import {IBeaconConfig} from "@chainsafe/lodestar-config";
+
+import {IDatabaseController} from "../../../controller";
+import {Bucket} from "../../schema";
+import {Repository} from "./abstract";
+import {ChainRepository} from "./chain";
+
+/**
+ * bad block roots
+ */
+export class BadBlockRepository extends Repository<Uint8Array, boolean> {
+
+  private chain: ChainRepository;
+
+  public constructor(
+    config: IBeaconConfig,
+    db: IDatabaseController<Buffer, Buffer>,
+  ) {
+    super(config, db, Bucket.invalidBlock, config.types.Boolean);
+  }
+
+  // eslint-disable-next-line @typescript-eslint/no-unused-vars
+  public getId(value: boolean): Uint8Array {
+    throw new Error("Cannot get block root from boolean");
```

Comment:
eh?

Reply:
this `getId` method infers the key from a value.  (its what helps transform `add(value)` and `remove(value)` methods, into `put(key, value)`, `delete(key, value)`)
In the case of bad block, the key is the block root, the value is a boolean. So its not possible to infer the key from the value there.